django/contrib/postgres/lookups.py

```diff
@@ -60,3 +60,24 @@ def process_lhs(self, qn, connection):
             self.lhs = SearchVector(self.lhs)
         lhs, lhs_params = super(SearchLookup, self).process_lhs(qn, connection)
         return lhs, lhs_params
+
+
+class TrigramSimilar(PostgresSimpleLookup):
+    lookup_name = 'trigram_similar'
+    operator = '%%'
+
+
+class TrigramDistance(PostgresSimpleLookup):
```

Comment:
> Just so I'm clear then, sorry, would filter(foo__trigram_distance=("Test", 0.5)) be the same as filter(foo=TrigramDistance("Test", 0.5))? If so, then that's fine, I just thought they differed somehow.

Until [#25367](https://code.djangoproject.com/ticket/25367) is fixed you have to rely on annotate. Should be along the lines of:

```
annotate(
    foo_distance=TrigramDistance(F('foo'), 'Test')
).filter(foo_distance__lte=0.5)`
```